src/App/SqsMessage.hs

```diff
@@ -36,13 +58,14 @@ decodeSqsMessage sqsMessage = do
   objectKey  <- objectAws ^. key "key"
   objectSize <- objectAws ^. key "size"
   -- there is `eTag` field in ObjectCreated:Putevent and no such field in ObjectCreated:Copy
-  let objectTag = fromMaybe "" (objectAws  ^. key "eTag")
-
-  return FileChangeMessage
-    { fileChangeMessageEventName  = eventName
-    , fileChangeMessageEventTime  = eventTime
-    , fileChangeMessageBucketName = bucketName
-    , fileChangeMessageObjectKey  = objectKey
-    , fileChangeMessageObjectSize = objectSize
-    , fileChangeMessageObjectTag  = objectTag
-    }
+  let objectTag = fromMaybe "" (objectAws ^. key "eTag")
+
+  return $ SqsMessageOfFileChangeMessage $
+    FileChangeMessage
+      { fileChangeMessageEventName  = eventName
+      , fileChangeMessageEventTime  = eventTime
+      , fileChangeMessageBucketName = bucketName
+      , fileChangeMessageObjectKey  = objectKey
+      , fileChangeMessageObjectSize = objectSize
+      , fileChangeMessageObjectTag  = objectTag
+      }
```

Comment:
We could do with some unit tests.  Can be in a subsequent PR?

Reply:
👍

Reply:
👍